### PR TITLE
made changes for new api and improved tests

### LINKED ISSUE #68: update to new API of DataFrames.jl
DataFrames.jl has released a new version.
Functions like `collect_results` throw lots of deprecation warnings now.

We should update this before errors start showing up.

### PR DESCRIPTION
**Pull request by JonasIsensee**
I made the relevant changes. No more deprecation warnings.

Also slightly improved the tests to cover more possibilities.

closes #68

### CODE REVIEW

**Comment by Datseris** [comment]
@tamasgal  or @asinghvi17  would you like to review this? Otherwise I'll do it.

**Comment by asinghvi17** [comment]
Just looked through the code briefly, but you should probably also review it @Datseris.  One question I had - is there a need for `.=`, or can you just wrap in an array and do `=`?

**Comment by JonasIsensee** [comment]
AFAICT the deprecation warnings of `DataFrames` precisely requested this syntax.
See also
https://discourse.julialang.org/t/release-announcements-for-dataframes-jl/18258/38

**Comment by Datseris** [comment]
Okay I'll try to review but I am clueless about DataFrames so... :D I'll try my best

File: src/result_collection.jl [hunk 1]
```diff
@@ -90,7 +90,7 @@ function collect_results!(filename, folder;
     end
 
     n = 0 # new entries added
-    existing_files = get(df, :path, ())
```
Comment (Datseris) [inline src/result_collection.jl]: I assume that `get(df, :path, ())` was deprecated? Too bad, the alternative `:path in names(df) ? df[:,:path] : ()` is unreasonably verbose :(

Comment (JonasIsensee) [reply src/result_collection.jl]: Sadly it was.

**Pull request merged**